### ISSUE TITLE
Alpaca Flatpak Version Fails with Connection Error During Chat

### ISSUE THREAD

**Issue by Xathros1** [Jeffser/Alpaca#157]
#### Describe the bug
After loading the model (gemma2:latest) in Alpaca on openSUSE Tumbleweed, attempting to use the chat feature results in a connection error. The local Ollama instance is reset, but the issue persists, causing repeated errors and preventing the chat from functioning properly. The error log indicates a `RemoteDisconnected` exception.

#### Expected behavior
The chat feature should function correctly, providing responses without causing connection errors or resetting the local Ollama instance.

#### Additional context
- Operating System: openSUSE Tumbleweed (20240716)
- Application Version: Alpaca 1.0.0 (Flatpak version)
- The issue occurs consistently across multiple attempts.
- The problem persists even after restarting the application and the operating system.
- Log output indicates repeated connection errors and exceptions related to `RemoteDisconnected`.

#### Log Output
```
cb@opensus-desktop:~> flatpak run com.jeffser.Alpaca
INFO    [main.py | main] Alpaca version: 1.0.0
INFO    [local_instance.py | start] Starting Alpaca's Ollama instance...
INFO    [local_instance.py | start] Started Alpaca's Ollama instance
[GIN] 2024/07/23 - 15:09:32 | 200 |      16.672µs |       127.0.0.1 | GET      "/"
[GIN] 2024/07/23 - 15:09:32 | 200 |     278.338µs |       127.0.0.1 | GET      "/api/tags"
INFO [main] build info | build=1 commit="74f33ad" tid="140418854479744" timestamp=1721740190
INFO [main] system info | n_threads=8 n_threads_batch=-1 system_info="AVX = 1 | AVX_VNNI = 0 | AVX2 = 1 | AVX512 = 0 | AVX512_VBMI = 0 | AVX512_VNNI = 0 | AVX512_BF16 = 0 | FMA = 1 | NEON = 0 | ARM_FMA = 0 | F16C = 1 | FP16_VA = 0 | WASM_SIMD = 0 | BLAS = 0 | SSE3 = 1 | SSSE3 = 1 | VSX = 0 | MATMUL_INT8 = 0 | LLAMAFILE = 1 | " tid="140418854479744" timestamp=1721740190 total_threads=16
INFO [main] HTTP server listening | hostname="127.0.0.1" n_threads_http="15" port="41517" tid="140418854479744" timestamp=1721740190
[GIN] 2024/07/23 - 15:09:51 | 500 |  1.211906536s |       127.0.0.1 | POST     "/api/chat"
ERROR   [window.py | connection_error] Connection error
INFO    [local_instance.py | reset] Resetting Alpaca's Ollama instance
INFO    [local_instance.py | stop] Stopping Alpaca's Ollama instance
INFO    [local_instance.py | stop] Stopped Alpaca's Ollama instance
INFO    [local_instance.py | start] Starting Alpaca's Ollama instance...
INFO    [local_instance.py | start] Started Alpaca's Ollama instance
INFO    [window.py | show_toast] There was an error with the local Ollama instance, so it has been reset
INFO [main] build info | build=1 commit="74f33ad" tid="140452920727424" timestamp=1721740216
INFO [main] system info | n_threads=8 n_threads_batch=-1 system_info="AVX = 1 | AVX_VNNI = 0 | AVX2 = 1 | AVX512 = 0 | AVX512_VBMI = 0 | AVX512_VNNI = 0 | AVX512_BF16 = 0 | FMA = 1 | NEON = 0 | ARM_FMA = 0 | F16C = 1 | FP16_VA = 0 | WASM_SIMD = 0 | BLAS = 0 | SSE3 = 1 | SSSE3 = 1 | VSX = 0 | MATMUL_INT8 = 0 | LLAMAFILE = 1 | " tid="140452920727424" timestamp=1721740216 total_threads=16
INFO [main] HTTP server listening | hostname="127.0.0.1" n_threads_http="15" port="39483" tid="140452920727424" timestamp=1721740216
[GIN] 2024/07/23 - 15:10:17 | 500 |   1.25938338s |       127.0.0.1 | POST     "/api/chat"
ERROR   [window.py | connection_error] Connection error
INFO    [local_instance.py | reset] Resetting Alpaca's Ollama instance
INFO    [local_instance.py | stop] Stopping Alpaca's Ollama instance
Exception in thread Thread-3 (generate_chat_title):
Traceback (most recent call last):
  File "/app/lib/python3.11/site-packages/urllib3/connectionpool.py", line 793, in urlopen
    response = self._make_request(
               ^^^^^^^^^^^^^^^^^^^
  File "/app/lib/python3.11/site-packages/urllib3/connectionpool.py", line 537, in _make_request
    response = conn.getresponse()
               ^^^^^^^^^^^^^^^^^^
  File "/app/lib/python3.11/site-packages/urllib3/connection.py", line 466, in getresponse
    httplib_response = super().getresponse()
                       ^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/http/client.py", line 1395, in getresponse
    response.begin()
  File "/usr/lib/python3.11/http/client.py", line 325, in begin
    version, status, reason = self._read_status()
                              ^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/http/client.py", line 294, in _read_status
    raise RemoteDisconnected("Remote end closed connection without"
http.client.RemoteDisconnected: Remote end closed connection without response

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/app/lib/python3.11/site-packages/requests/adapters.py", line 486, in send
    resp = conn.urlopen(
           ^^^^^^^^^^^^^
  File "/app/lib/python3.11/site-packages/urllib3/connectionpool.py", line 847, in urlopen
    retries = retries.increment(
              ^^^^^^^^^^^^^^^^^^
  File "/app/lib/python3.11/site-packages/urllib3/util/retry.py", line 470, in increment
    raise reraise(type(error), error, _stacktrace)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/app/lib/python3.11/site-packages/urllib3/util/util.py", line 38, in reraise
    raise value.with_traceback(tb)
  File "/app/lib/python3.11/site-packages/urllib3/connectionpool.py", line 793, in urlopen
    response = self._make_request(
               ^^^^^^^^^^^^^^^^^^^
  File "/app/lib/python3.11/site-packages/urllib3/connectionpool.py", line 537, in _make_request
    response = conn.getresponse()
               ^^^^^^^^^^^^^^^^^^
  File "/app/lib/python3.11/site-packages/urllib3/connection.py", line 466, in getresponse
    httplib_response = super().getresponse()
                       ^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/http/client.py", line 1395, in getresponse
    response.begin()
  File "/usr/lib/python3.11/http/client.py", line 325, in begin
    version, status, reason = self._read_status()
                              ^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/http/client.py", line 294, in _read_status
    raise RemoteDisconnected("Remote end closed connection without"
urllib3.exceptions.ProtocolError: ('Connection aborted.', RemoteDisconnected('Remote end closed connection without response'))

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/lib/python3.11/threading.py", line 1045, in _bootstrap_inner
    self.run()
  File "/usr/lib/python3.11/threading.py", line 982, in run
    self._target(*self._args, **self._kwargs)
  File "/app/share/Alpaca/alpaca/window.py", line 578, in generate_chat_title
    response = connection_handler.simple_post(f"{connection_handler.url}/api/generate", data=json.dumps(data))
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/app/share/Alpaca/alpaca/connection_handler.py", line 19, in simple_post
    return requests.post(connection_url, headers=get_headers(True), data=data, stream=False)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/app/lib/python3.11/site-packages/requests/api.py", line 115, in post


    return request("post", url, data=data, json=json, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/app/lib/python3.11/site-packages/requests/api.py", line 59, in request
    return session.request(method=method, url=url, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/app/lib/python3.11/site-packages/requests/sessions.py", line 589, in request
    resp = self.send(prep, **send_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/app/lib/python3.11/site-packages/requests/sessions.py", line 703, in send
    r = adapter.send(request, **kwargs)
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/app/lib/python3.11/site-packages/requests/adapters.py", line 501, in send
    raise ConnectionError(err, request=request)
requests.exceptions.ConnectionError: ('Connection aborted.', RemoteDisconnected('Remote end closed connection without response'))
INFO    [local_instance.py | stop] Stopped Alpaca's Ollama instance
INFO    [local_instance.py | start] Starting Alpaca's Ollama instance...
INFO    [local_instance.py | start] Started Alpaca's Ollama instance
INFO    [window.py | show_toast] There was an error with the local Ollama instance, so it has been reset
```

**Comment by Jeffser** [comment]
Hi thanks for the report, maybe you have a different app using the port 11435, this is the port that is used by Alpaca, to change it go to `~/.var/app/com.jeffser.Alpaca/config/server.json` and change the value of `local_port`

**Comment by Xathros1** [comment]
Hi,

Thanks for the quick reply! I checked and made sure no other apps are using port 11435 on my system. I also changed the port number in `~/.var/app/com.jeffser.Alpaca/config/server.json` as you suggested, but unfortunately, the error is still the same.

**Comment by Jeffser** [comment]
honestly this is the first time I've seen an error like this, it seems like the Ollama instance doesn't want to respond to a chat request, could you try pulling a different model and see if that fixes it? thanks

**Comment by Xathros1** [comment]
I tried llama3:latest and that one works

**Comment by GaPhilippone** [comment]
Hello there, juste to add: 
llama3 work. 
DeepSeekCoder V2 doesn't work.
When running llama3 and asking something, changing to deepseek kill immediatly the ollama-server in the task manager

`flatpak run com.jeffser.Alpaca
INFO	[main.py | main] Alpaca version: 1.0.1
INFO	[local_instance.py | start] Starting Alpaca's Ollama instance...
INFO	[local_instance.py | start] Started Alpaca's Ollama instance
[GIN] 2024/07/27 - 06:26:49 | 200 |     814.483µs |       127.0.0.1 | GET      "/api/tags"
[GIN] 2024/07/27 - 06:26:49 | 200 |    1.095855ms |       127.0.0.1 | GET      "/api/tags"
INFO [main] build info | build=1 commit="74f33ad" tid="131582179307520" timestamp=1722054416
INFO [main] system info | n_threads=8 n_threads_batch=-1 system_info="AVX = 1 | AVX_VNNI = 0 | AVX2 = 0 | AVX512 = 0 | AVX512_VBMI = 0 | AVX512_VNNI = 0 | AVX512_BF16 = 0 | FMA = 0 | NEON = 0 | ARM_FMA = 0 | F16C = 0 | FP16_VA = 0 | WASM_SIMD = 0 | BLAS = 1 | SSE3 = 1 | SSSE3 = 1 | VSX = 0 | MATMUL_INT8 = 0 | LLAMAFILE = 1 | " tid="131582179307520" timestamp=1722054416 total_threads=8
INFO [main] HTTP server listening | hostname="127.0.0.1" n_threads_http="7" port="37643" tid="131582179307520" timestamp=1722054416
[GIN] 2024/07/27 - 06:26:56 | 500 |  2.197831193s |       127.0.0.1 | POST     "/api/chat"
ERROR	[window.py | connection_error] Connection error
INFO	[local_instance.py | reset] Resetting Alpaca's Ollama instance
INFO	[local_instance.py | stop] Stopping Alpaca's Ollama instance
INFO	[local_instance.py | stop] Stopped Alpaca's Ollama instance
INFO	[local_instance.py | start] Starting Alpaca's Ollama instance...
INFO	[local_instance.py | start] Started Alpaca's Ollama instance
INFO	[window.py | show_toast] There was an error with the local Ollama instance, so it has been reset
[GIN] 2024/07/27 - 06:27:39 | 200 |     896.291µs |       127.0.0.1 | GET      "/api/tags"
[GIN] 2024/07/27 - 06:27:40 | 200 |    1.056634ms |       127.0.0.1 | GET      "/api/tags"
INFO	[window.py | pull_model] Pulling model
[GIN] 2024/07/27 - 06:29:56 | 200 |         1m35s |       127.0.0.1 | POST     "/api/pull"
[GIN] 2024/07/27 - 06:29:56 | 200 |    1.093547ms |       127.0.0.1 | GET      "/api/tags"
INFO	[window.py | show_notification] Task Complete, Model 'llama3:latest' pulled successfully.
INFO	[window.py | show_toast] Model 'llama3:latest' pulled successfully.
INFO	[window.py | delete_chat] Deleting chat

(alpaca:2): Gtk-WARNING **: 06:30:15.205: Finalizing  0x600efd7304e0, but it still has children left:
   - GtkPopoverMenu 0x600efedcd210
INFO	[window.py | delete_chat] Deleting chat
llama3:latest
INFO [main] build info | build=1 commit="74f33ad" tid="135469791973376" timestamp=1722054623
INFO [main] system info | n_threads=8 n_threads_batch=-1 system_info="AVX = 1 | AVX_VNNI = 0 | AVX2 = 0 | AVX512 = 0 | AVX512_VBMI = 0 | AVX512_VNNI = 0 | AVX512_BF16 = 0 | FMA = 0 | NEON = 0 | ARM_FMA = 0 | F16C = 0 | FP16_VA = 0 | WASM_SIMD = 0 | BLAS = 1 | SSE3 = 1 | SSSE3 = 1 | VSX = 0 | MATMUL_INT8 = 0 | LLAMAFILE = 1 | " tid="135469791973376" timestamp=1722054623 total_threads=8
INFO [main] HTTP server listening | hostname="127.0.0.1" n_threads_http="7" port="38283" tid="135469791973376" timestamp=1722054623
INFO [main] model loaded | tid="135469791973376" timestamp=1722054629
[GIN] 2024/07/27 - 06:30:38 | 200 | 18.762863571s |       127.0.0.1 | POST     "/api/generate"
INFO	[window.py | rename_chat] Renaming chat "New Chat" -> "Test Title"
[GIN] 2024/07/27 - 06:30:57 | 200 |  37.00784907s |       127.0.0.1 | POST     "/api/chat"
INFO	[window.py | show_notification] Test Title, It looks like you're just testing the waters here! That's perfectly fine. I'm happy to chat and resp...
INFO [main] build info | build=1 commit="74f33ad" tid="139733219794944" timestamp=1722054680
INFO [main] system info | n_threads=8 n_threads_batch=-1 system_info="AVX = 1 | AVX_VNNI = 0 | AVX2 = 0 | AVX512 = 0 | AVX512_VBMI = 0 | AVX512_VNNI = 0 | AVX512_BF16 = 0 | FMA = 0 | NEON = 0 | ARM_FMA = 0 | F16C = 0 | FP16_VA = 0 | WASM_SIMD = 0 | BLAS = 1 | SSE3 = 1 | SSSE3 = 1 | VSX = 0 | MATMUL_INT8 = 0 | LLAMAFILE = 1 | " tid="139733219794944" timestamp=1722054680 total_threads=8
INFO [main] HTTP server listening | hostname="127.0.0.1" n_threads_http="7" port="41809" tid="139733219794944" timestamp=1722054680
[GIN] 2024/07/27 - 06:31:20 | 500 |  2.635362008s |       127.0.0.1 | POST     "/api/chat"
ERROR	[window.py | connection_error] Connection error
INFO	[local_instance.py | reset] Resetting Alpaca's Ollama instance
INFO	[local_instance.py | stop] Stopping Alpaca's Ollama instance
INFO	[local_instance.py | stop] Stopped Alpaca's Ollama instance
INFO	[local_instance.py | start] Starting Alpaca's Ollama instance...
INFO	[local_instance.py | start] Started Alpaca's Ollama instance
INFO	[window.py | show_toast] There was an error with the local Ollama instance, so it has been reset
`

**Comment by Jeffser** [comment]
Hi, sorry for the delay, I just figured out what was wrong.

So, Ollama is trying to extract itself into the tmp folder, Ollama by itself is really big and sometimes and it doesn't fit inside the limit for tmp that some distros set, I moved Ollama to the cache folder of Alpaca which gets deleted everytime the app is launched, this should fix the issue for 1.0.2, if not, please reopen the issue!

**Comment by Xathros1** [comment]
![Bildschirmfoto_20240729_141725](https://github.com/user-attachments/assets/c8908f5f-a8da-4779-a59d-7dc2c61862e9)
can confirm this fixes the issue for me
